homeassistant/components/recorder/migration.py

```diff
@@ -41,20 +41,39 @@ def _apply_update(engine, new_version):
     from sqlalchemy import Table
     from . import models
 
-    if new_version == 1:
-        def create_index(table_name, column_name):
-            """Create an index for the specified table and column."""
-            table = Table(table_name, models.Base.metadata)
-            name = "_".join(("ix", table_name, column_name))
-            # Look up the index object that was created from the models
-            index = next(idx for idx in table.indexes if idx.name == name)
-            _LOGGER.debug("Creating index for table %s column %s",
-                          table_name, column_name)
-            index.create(engine)
-            _LOGGER.debug("Index creation done for table %s column %s",
-                          table_name, column_name)
+    def create_index(table_name, column_name):
+        """Create an index for the specified table and column."""
+        table = Table(table_name, models.Base.metadata)
+        name = "_".join(("ix", table_name, column_name))
```

Comment:
What about something like this to combine these into one function? The pattern is usually called `*args` if you need to google it. You can replace the debug messages with the one you wrote for compound index so the individual column names don't need to be used.

```python
def create_index(table_name, *column_names):
   name = "_".join("_".join("ix", table_name), column_names)
    ...
```

Then it could be used for both index types like this
```python
create_index("recorder_runs", "start", "end")
create_index("states", "last_updated")
```